src/selection/highlight/highlight.js

```diff
@@ -90,12 +90,14 @@ class Highlight {
    * Check if highlight cell rendering is disabled for specified highlight type.
    *
    * @param {string} highlightType Highlight type. Possible values are: `cell`, `area`, `fill` or `header`.
+   * @param {CellCoords} coords The CellCoords instance with defined visual coordinates.
    * @returns {boolean}
    */
-  isEnabledFor(highlightType) {
+  isEnabledFor(highlightType, coords) {
     // Legacy compatibility.
     const type = highlightType === 'current' ? CELL_TYPE : highlightType;
-    let disableHighlight = this.options.disableHighlight;
+
+    let disableHighlight = this.options.disabledCellSelection(coords.row, coords.col);
```

Comment:
Given:

```js
cells(row, column) {
	return {
		disableVisualSelection: row === 1 && column === 1
	}
}
```

...only starting the selection on (1,1) will actually disable the visual selection (and will so for the whole table).

I expected something like this to happen instead:

<img width="277" alt="image" src="https://user-images.githubusercontent.com/7091471/105330980-3330df00-5bd3-11eb-870c-0af6ac04a867.png">

(same with columns and rows)

Why is the implemented behavior the intended one, rather than what I sent? It doesn't make sense to me at the moment.

Reply:
I can agree with you that it may be strange. We may change it within another issue as the original one is mainly focused on regression. 

It seems that this behaviour is unexpectedly intended as it worked in that way also for Handsontable <`0.36.0`. Thus, any change should be discussed with the UI/UX first.